### PR TITLE
Create reservation

### PR DESCRIPTION
**Pull request by Sponken**

### CODE REVIEW

File: .idea/compiler.xml [hunk 1]
```diff
@@ -6,12 +6,14 @@
...
         <sourceTestOutputDir name="target/generated-test-sources/test-annotations" />
         <outputRelativeToContentRoot value="true" />
+        <module name="tda367-owls" />
         <module name="StoreIT" />
       </profile>
     </annotationProcessing>
     <bytecodeTargetLevel>
       <module name="OwlsOOPProject" target="1.8" />
       <module name="StoreIT" target="1.8" />
+      <module name="tda367-owls" target="1.8" />
```
Comment (steget) [inline .idea/compiler.xml]: ne

File: src/main/java/storagesystem/model/Organisation.java [hunk 2]
```diff
@@ -111,6 +112,25 @@ public boolean isUserPartOfTeam(IBorrower user){
...
     }
 
+
+    public Team getTeamFromName(String teamName){
+        for(Team t : teams){
+            if (t.getName().equals(teamName)){
+                return t;
+            }
+        }
+        return null;
```
Comment (steget) [inline src/main/java/storagesystem/model/Organisation.java]: throw some error

File: src/main/java/storagesystem/viewcontroller/basicStyles.css [hunk 3]
```diff
@@ -98,7 +116,28 @@
...
 
+.confirmButton{
+    -fx-background-radius: 10;
+    -fx-background-color: appButtonColor;
+    -fx-font-size: 20px;
+    -fx-text-fill: white;
+}
+
+.confirmButton:hover{
+    /*TODO Change color to match other buttons*/
```
Comment (steget) [inline src/main/java/storagesystem/viewcontroller/basicStyles.css]: do it

File: src/main/java/storagesystem/viewcontroller/inventory/ItemDetailViewController.java [hunk 4]
```diff
@@ -171,7 +178,9 @@ private void setReservableChoiceBox() {
 
     @FXML
     protected void reserveBtnPressed() {
-        //TODO: create a new reservation if possible
+        for(ReserveButtonClickedListener listener : reserveButtonClickedListeners){
+            listener.reserveButtonClicked();
+        }
```
Comment (steget) [inline src/main/java/storagesystem/viewcontroller/inventory/ItemDetailViewController.java]: add todo to move this to all items

File: src/main/java/storagesystem/viewcontroller/reservations/CreateReservationController.java [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+        startYearSpinner.setValueFactory(startYearFactory);
+        endYearSpinner.setValueFactory(endYearFactory);
+        startMonthSpinner.setValueFactory(startMonthFactory);
+        endMonthSpinner.setValueFactory(endMonthFactory);
+        startDaySpinner.setValueFactory(startDayFactory);
+        endDaySpinner.setValueFactory(endDayFactory);
+        startHourSpinner.setValueFactory(startHourFactory);
+        endHourSpinner.setValueFactory(endHourFactory);
+        startMinuteSpinner.setValueFactory(startMinuteFactory);
+        endMinuteSpinner.setValueFactory(endMinuteFactory);
```
Comment (steget) [inline src/main/java/storagesystem/viewcontroller/reservations/CreateReservationController.java]: move all of this to its own method?

File: src/main/java/storagesystem/viewcontroller/reservations/CreateReservationController.java [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+
+        this.item = item;
+
+    }
+
+
+    @FXML
+    private void confirm() {
```
Comment (steget) [inline src/main/java/storagesystem/viewcontroller/reservations/CreateReservationController.java]: confirm what? Better method name

File: src/main/java/storagesystem/viewcontroller/reservations/CreateReservationController.java [hunk 7]
```diff
@@ -0,0 +1,221 @@
...
+        int endHour = endHourSpinner.getValue();
+        int endMinute = endMinuteSpinner.getValue();
+        DateTime start = new DateTime(startYear, startMonth, startDay, startHour, startMinute);
+        DateTime end = new DateTime(endYear, endMonth, endDay, endHour, endMinute);
+
+        return new Interval(start, end);
+    }
+
+    @FXML
+    private void cancel() {
```
Comment (steget) [inline src/main/java/storagesystem/viewcontroller/reservations/CreateReservationController.java]: why does this method exist?

File: src/main/java/storagesystem/viewcontroller/basicStyles.css (hunk 3 above)
Comment (Sponken) [reply src/main/java/storagesystem/viewcontroller/basicStyles.css]: No, am waiting for commit from another branch

File: src/main/java/storagesystem/viewcontroller/inventory/ItemDetailViewController.java [hunk 8]
```diff
@@ -252,8 +251,8 @@ public void addSaveButtonListener(saveButtonClickedListener listener) {
      * The default setting is that the variables are non-editable.
      */
     public void editItem() {
-
-        itemPageReserveBtn.setVisible(false);
+        //TODO make not visible
+        //itemPageReserveBtn.setVisible(false);
```
[the remark this answers is not in the document]
Comment (Sponken) [reply src/main/java/storagesystem/viewcontroller/inventory/ItemDetailViewController.java]: Yes, not my problem to fix

File: src/main/java/storagesystem/model/Team.java [hunk 9]
```diff
@@ -9,13 +9,13 @@
  *
  * @author Hugo Stegrell, Pär Aronsson
  */
-public class Team {
-    private static int nextID;
+public class Team implements IBorrower{
+    private String name;
     private final List<IReservable> inventory = new ArrayList<>(); //todo itemIDs instead
```
[the remark this answers is not in the document]
Comment (Sponken) [reply src/main/java/storagesystem/model/Team.java]: Haven't merged properly it seems, fixed soon

File: src/main/java/storagesystem/model/Team.java [hunk 10]
```diff
@@ -9,13 +9,13 @@
...
     private final List<Integer> memberIDs = new ArrayList<>();
-    private String name;
     private String termsAndConditions;
     private int teamID;
```
[the remark this answers is not in the document]
Comment (Kviick) [reply src/main/java/storagesystem/model/Team.java]: Still not changed teamID to ID (no use to name it teamID since the variable is in Team)

**Pull request merged**